### PR TITLE
Provider request reduction

### LINKED ISSUE #625: Understand why so many RPCs are being made by agents, and ideally reduce unnecessary ones
We're noticing each endpoint is being hit with a couple hundred thousand RPCs per day, mostly `eth_call`s, which is more than we'd expect. We should investigate to understand what code is responsible for all these RPCs, determine which (if any) frequencies should be toned down, and make the necessary changes.

A good jumping off point is to check out QuickNode / Ankr / Alchemy and look at the metrics of endpoints, which usually has a breakdown of the types of RPCs. e.g. QuickNode's Polygon/Matic endpoint has a good breakdown of them. There may be helpful metrics in Grafana to take a look at.

Determining which calls are eligible to be reduced may be a hard call to make, so figuring out the sources of each call and then discussing them with the team is also a good first step

### LINKED ISSUE #632: Gauge metrics should be set on agent start
Currently having an issue where one of the pods restarted and it caused the latest validated index to go from 5 -> 0, which means it looks like the validators are 5 messages behind, when really it was just a pod restarting and the gauge initializing at zero.

### PR DESCRIPTION
**Pull request by mattiecnvr**
- Reduces the number of calls made to the providers. For both relayers and validators this switches to querying outbox state every 60s rather than doing it as part of the loop
- Adds two new metrics to track contract calls down to the function
- Removes the old contract call duration histogram
- Now initializes the latest index processed gauge

Closes #625 
Closes #632

### CODE REVIEW

File: rust/abacus-base/src/oneline_eyre/handler.rs [hunk 1]
```diff
@@ -86,6 +86,7 @@ pub struct HookBuilder {
     show_full_paths: bool,
 }
 
+#[allow(dead_code)]
```
Comment (webbhorn) [inline rust/abacus-base/src/oneline_eyre/handler.rs:89]: why?

**Review by webbhorn — APPROVED** [review]
Looks good, my only high level concern (which I have high uncertainty about and suspect doesn't matter), is around the scenario of function signatures changing on any contracts, if that's even allowed, and whether that would introduce any undesired discontinuities in our timeseries data

File: rust/abacus-base/src/oneline_eyre/handler.rs (hunk 1 above)
Comment (mattiecnvr) [reply rust/abacus-base/src/oneline_eyre/handler.rs:89]: Because it is a builder, not using a function now does not mean it should not be there.

**Comment by mattiecnvr** [comment]
@webbhorn That's a good concern, this is why I use names as well. The hashes are mostly there for debugging.

**Pull request merged**